Submitted By: Alexander E. Patrakov
Date: 2006-04-05
Initial Package Version: 0.8.10
Upstream Status: ???
Origin: http://people.ubuntu.com/~scott/patches/irssi/irssi_0.8.10-1ubuntu1.patch
Description: Fixes segfault-on-quit with Glib >= 2.10

diff -urNad irssi-0.8.10~/src/irc/dcc/dcc.c irssi-0.8.10/src/irc/dcc/dcc.c
--- irssi-0.8.10~/src/irc/dcc/dcc.c	2006-01-18 14:25:19.000000000 +0200
+++ irssi-0.8.10/src/irc/dcc/dcc.c	2006-01-18 14:41:20.000000000 +0200
@@ -58,8 +58,8 @@
 
 	pos = gslist_find_string(dcc_types, type);
 	if (pos != NULL) {
-                dcc_types = g_slist_remove(dcc_types, pos->data);
 		g_free(pos->data);
+                dcc_types = g_slist_remove(dcc_types, pos->data);
 	}
 }
 
